### DISCUSSION TITLE
How Do I Obtain the Memory Size of the lyd_node?

### DISCUSSION THREAD

**Discussion by zhangtaog** [CESNET/libyang#2296] [Q&A] [answered]
I need to collect statistics on the memory usage of a single lyd_node or the entire tree node. Is there any method to obtain or calculate the memory usage? Thank you！

**Comment by michalvasko** [comment, marked as answer]
Generally, you should use a memory measuring tool such as `massif`. But to answer, you get the memory usage of a single `struct lyd_node *node` by calling `sizeof(*node)`.
